### PR TITLE
Sambamba depth recipe

### PR DESCRIPTION
**Pull request by northwestwitch**
Added Sambamba depth recipe

### CODE REVIEW

File: lib/MIP/Recipes/Sambamba_depth.pm [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+sub analysis_sambamba_depth {
+
+## Function : Generate coverage bed outfile for each individual.
+## Returns  : ""
+## Arguments: $parameter_href, $active_parameter_href, $sample_info_href, $file_info_href, $infile_lane_prefix_href, $job_id_href, $sample_id, $program_name, family_id, $temp_directory, $outaligner_dir
+##          : $parameter_href             => Parameter hash {REF}
+##          : $active_parameter_href      => Active parameters for this analysis hash {REF}
+##          : $sample_info_href           => Info on samples and family hash {REF}
+##          : $file_info_href             => The file_info hash {REF}
+##          : $infile_lane_prefix_href    => Infile(s) without the ".ending" {REF}
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Sambamba_depth.pm]: Align fat commas: $infile_lane_prefix_href    => -> $infile_lane_prefix_href =>

File: lib/MIP/Recipes/Sambamba_depth.pm [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+##          : $job_id_href                => Job id hash {REF}
+##          : $sample_id                  => Sample id
+##          : $outaligner_dir             => Outaligner_dir used in the analysis
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Sambamba_depth.pm]: Present twice

File: lib/MIP/Recipes/Sambamba_depth.pm [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+    my $core_number =
+      $active_parameter_href->{module_core_number}{$mip_program_name};
+    my $time = $active_parameter_href->{module_time}{$mip_program_name};
+
+    ## Filehandles
+    # Create anonymous filehandle
+    my $FILEHANDLE = IO::Handle->new();
+
+    ## Assign directories
+    my $insample_directory = catdir( $active_parameter_href->{outdata_dir},
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Sambamba_depth.pm]: Move these (in/out dirs) to be part of the sub call. This makes it more apparent what goes into the sub and what comes out

File: lib/MIP/Recipes/Sambamba_depth.pm [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+            outfile_path => $temp_directory
+        }
+    );
+    say {$FILEHANDLE} q{wait}, $NEWLINE;
+
+    ## sambamba_depth
+    say {$FILEHANDLE} q{## Annotating bed from alignment};
+
+    ## Get parameters
+    my $sambamba_filter = q?'mapping_quality >= ?
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Sambamba_depth.pm]: q?? -> q{ }
and
q? ? -> $SPACE

File: lib/MIP/Recipes/Sambamba_depth.pm [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+
+        $sambamba_filter .= q{and not duplicate };
+    }
+
+    #Do not include failed quality control reads in coverage calculation
+    if ( $active_parameter_href->{sambamba_depth_quality_control} ) {
+
+        $sambamba_filter .= q{and not failed_quality_control};
+    }
+    $sambamba_filter .= q?'?;
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Sambamba_depth.pm]: q?'? -> q{'}

File: lib/MIP/Recipes/Sambamba_depth.pm (hunk 3 above)
Comment (northwestwitch) [reply lib/MIP/Recipes/Sambamba_depth.pm]: The migrate_file sub you mean?

File: lib/MIP/Recipes/Sambamba_depth.pm (hunk 1 above)
Comment (henrikstranneheim) [reply lib/MIP/Recipes/Sambamba_depth.pm]: Do not forget to align the fat commas

**Comment by henrikstranneheim** [comment]
Great!

**Pull request merged**